packages/flutter/lib/src/widgets/bottom_navigation_bar_item.dart

```diff
@@ -23,13 +23,17 @@ class BottomNavigationBarItem {
   ///
   /// The argument [icon] should not be null and the argument [label] should not be null when used in a Material Design's [BottomNavigationBar].
   const BottomNavigationBarItem({
+    this.key,
     required this.icon,
     this.label,
     Widget? activeIcon,
     this.backgroundColor,
     this.tooltip,
   }) : activeIcon = activeIcon ?? icon;
 
+  /// A key to be passed through to the _BottomNavigationTile widget
```

Comment:
Instead of referencing a private API here, can you explain what the key is used for, and an example use case (doesn't have to be a code example, just a use case, although I wouldn't turn a code example down. :-) ).

Reply:
thanks for the review, have updated. Hope this is what you were after :)